SWIPs/swip-streaming-of-feeds.md

```diff
@@ -0,0 +1,148 @@
+The lookup time of feeds can be significantly long where the lookup process only stops if there is no newer content under a feed. 
+
+A solution is needed where this approach is reversed: a lookup that stops if there is a successful hit.
+The lookup time can be shorthened already by not waiting for the last (non-existing) feed retrieval.
+
+If the uploader keeps itself to the periodic feed indexing and it uploads every time when is needed, the retrieval for the user is `O(1)`.
+Nevertheless, the lookups can easily go wrong, because of (1) network issues or (2) the uploader cannot upload the content in time.
+These problems are equivalent from the retrieval perspective and the first one is also related to the epoch-based and sequential feeds. 
+
+Any chunk of the lookup trajectory is not available (even temporary) then it could cause huge inaccuracy, but an approach like this is always closer or the same close to a desired upload time.
+
+Moreover, the current lookup methods keep those chunks alive that maybe unnecessary from the content usage perspective. 
+
+One use-case is the feed stream of the content (or dApp) is for providing its most recent state.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
+
+In the following subsections I would like to detail how to:
+* [get the nearest last index of an arbitrary time](###-Nearest-Last-Index-for-an-Arbitrary-Time)
+* [construct feed topic](###-Feed-Topic-Construction)
+* [upload feed chunk](###-Upload-Feed-Chunk)
+* [download feed chunk at specific time](###-Download-Feed-Chunk-At-Specific-Time)
+* [and download (whole) feed stream](###-Download-Feed-Stream)
+
+The feeds always have a `topic` and an `index`, that we have to define.
+
+### Nearest Last Index for an Arbitrary Time
+
+Let's figure out what is the nearest last `feed index` of a given arbitrary time.
+We know the current time right now (`Tp`) which is surely greater or equal to the last upload time (`Tn`) of the feed stream.
+We also know two metadata of the stream: initial time (`T0`) and update period (`Δ1`).
+We want to find the nearest last index to an arbitrary time (`Tx`) where `T0 <= Tx <= Tn <= Tp`
+All of these are timestamps and their smallest unit is 1 second, e.g. even with this time unit it is still uncertain whether the latest update can be downloaded if `Tp = Tn`, because of the nature of P2P storage systems.
+
+So the formula is really simple how to calculate the nearest last index parameter (`i`) of the `Tx`:
+
+```ts
+function getIndexForArbitraryTime(Tx: number, T0: number, updatePeriod: number): number {
+  return Math.floor((Tx - T0) / updatePeriod) // which is the `i`
+}
+```
```

Comment:
I aggree, I wanted to refer to the `UNIX Epoch Time` by `timestamps` in the text, but that phrasing of mine is really not that concrete, I will change that.
Regarding the delta should be power based 2, I have some concerns about this because it restricts the arbitrary time upload period that the uploader was free to choose in this original concept.
Suppose I want to set my delta to 86400 seconds (1 day), then I can easily set my `cron job` to upload feeds with this time period, but on the other hand I cannot do that with time period of power based 2.